disttask/framework/scheduler/manager.go

```diff
@@ -0,0 +1,307 @@
+	schedulerPool   Pool
+	// taskType -> subtaskExecutorPool
+	subtaskExecutorPools map[string]Pool
+	mu                   struct {
+		sync.RWMutex
+		// taskID -> cancelFunc
+		// cancelFunc is used to fast cancel the scheduler.Run
+		handlingTasks map[int64]context.CancelFunc
+	}
+	id     string
+	wg     sync.WaitGroup
+	ctx    context.Context
+	cancel context.CancelFunc
+	logCtx context.Context
+}
+
+// NewManager creates a new Manager.
+func NewManager(ctx context.Context, id string, globalTaskTable TaskTable, subtaskTable SubtaskTable) (*Manager, error) {
+	m := &Manager{
+		id:                   id,
+		globalTaskTable:      globalTaskTable,
+		subtaskTable:         subtaskTable,
+		subtaskExecutorPools: make(map[string]Pool),
+		logCtx:               logutil.WithKeyValue(context.Background(), "dist_task_manager", id),
+	}
+	m.ctx, m.cancel = context.WithCancel(ctx)
+	m.mu.handlingTasks = make(map[int64]context.CancelFunc)
+
+	schedulerPool, err := newPool("scheduler_pool", schedulerPoolSize, util.DDL)
+	if err != nil {
+		return nil, err
+	}
+	m.schedulerPool = schedulerPool
+
+	for taskType := range subtaskExecutorConstructors {
+		poolSize := subtaskExecutorPoolSize
+		if opt, ok := subtaskExecutorOptions[taskType]; ok && opt.PoolSize > 0 {
+			poolSize = opt.PoolSize
+		}
+		subtaskExecutorPool, err := newPool(taskType+"_pool", poolSize, util.DDL)
+		if err != nil {
+			return nil, err
+		}
+		m.subtaskExecutorPools[taskType] = subtaskExecutorPool
+	}
+	return m, nil
+}
+
+// Start starts the Manager.
+func (m *Manager) Start() {
+	m.wg.Add(1)
+	go func() {
+		defer m.wg.Done()
+		m.fetchAndHandleRunnableTasks(m.ctx)
+	}()
+
+	m.wg.Add(1)
+	go func() {
+		defer m.wg.Done()
+		m.fetchAndFastCancelTasks(m.ctx)
+	}()
+}
+
+// Stop stops the Manager.
+func (m *Manager) Stop() {
+	m.cancel()
+	for _, pool := range m.subtaskExecutorPools {
+		pool.ReleaseAndWait()
+	}
+	m.schedulerPool.ReleaseAndWait()
+	m.wg.Wait()
+}
+
+// fetchAndHandleRunnableTasks fetches the runnable tasks from the global task table and handles them.
+func (m *Manager) fetchAndHandleRunnableTasks(ctx context.Context) {
+	ticker := time.NewTicker(checkTime)
+	for {
+		select {
+		case <-ctx.Done():
+			logutil.Logger(m.logCtx).Info("fetchAndHandleRunnableTasks done")
+			return
+		case <-ticker.C:
+			tasks, err := m.globalTaskTable.GetTasksInStates(proto.TaskStateRunning, proto.TaskStateReverting)
+			if err != nil {
+				m.onError(err)
+				continue
+			}
+			m.onRunnableTasks(ctx, tasks)
+		}
+	}
+}
+
+// fetchAndFastCancelTasks fetches the reverting tasks from the global task table and fast cancels them.
+func (m *Manager) fetchAndFastCancelTasks(ctx context.Context) {
+	ticker := time.NewTicker(checkTime)
+	for {
+		select {
+		case <-ctx.Done():
+			m.cancelAllRunningTasks()
+			logutil.Logger(m.logCtx).Info("fetchAndFastCancelTasks done")
+			return
+		case <-ticker.C:
+			tasks, err := m.globalTaskTable.GetTasksInStates(proto.TaskStateReverting)
```

Comment:
If we get the `TaskStateReverting` task in `fetchAndHandleRunnableTasks` and prepare to do something rollback. 
Then here we get this task, and cancel the task, is it expected?